### PR TITLE
Add command callback table support

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR targets #365. It is WIP but I'm ready for review here.

### CODE REVIEW

File: src/compositor/framebuffer_device.c [hunk 1]
```diff
@@ -74,7 +74,10 @@ ws_object_type_id WS_OBJECT_TYPE_ID_FRAMEBUFFER_DEVICE = {
     .hash_callback = device_hash,
     .dump_callback = NULL,
     .run_callback = NULL,
-    .cmp_callback = device_cmp
+    .cmp_callback = device_cmp,
+    .cmd_callback = NULL,
```
Comment (neithernut) [inline src/compositor/framebuffer_device.c]: Please fix this.

File: src/compositor/monitor.c [hunk 2]
```diff
@@ -94,7 +94,10 @@ ws_object_type_id WS_OBJECT_TYPE_ID_MONITOR = {
     .hash_callback = monitor_hash,
     .dump_callback = NULL,
     .run_callback = NULL,
-    .cmp_callback = monitor_cmp
+    .cmp_callback = monitor_cmp,
+    .cmd_callback = NULL,
```
Comment (neithernut) [inline src/compositor/monitor.c]: Please fix this.

File: src/compositor/monitor_mode.c [hunk 3]
```diff
@@ -59,7 +59,10 @@ ws_object_type_id WS_OBJECT_TYPE_ID_MONITOR_MODE = {
     .hash_callback = monitor_mode_hash,
     .dump_callback = NULL,
     .run_callback = NULL,
-    .cmp_callback = monitor_mode_cmp
+    .cmp_callback = monitor_mode_cmp,
+    .cmd_callback = NULL,
```
Comment (neithernut) [inline src/compositor/monitor_mode.c]: Please fix this.

File: src/objects/object.c [hunk 4]
```diff
@@ -598,6 +600,27 @@ ws_object_is_instance_of(
...
+    ws_regular_command_func func = NULL;
+    struct ws_object_function const* iter;
+    for (iter = self->id->function_table; func == NULL && iter; iter++) {
+        if (strcmp(iter->name, ident) == 0) {
+            func = iter->func;
+        }
+    }
+
+    return func(stack);
+}
```
Comment (neithernut) [inline src/objects/object.c:628]: This will segfault if no function was found.

**Comment by neithernut** [comment]
You have to rebase.

File: src/objects/object.c [hunk 5]
```diff
@@ -597,6 +598,31 @@ ws_object_is_instance_of(
...
+        return -EINVAL;
+    }
+
+    ws_regular_command_func func = NULL;
+    struct ws_object_function const* iter;
+    for (iter = self->id->function_table; func == NULL && iter; iter++) {
+        if (strcmp(iter->name, ident) == 0) {
+            func = iter->func;
+        }
+    }
```
Comment (neithernut) [inline src/objects/object.c:621]: `iter` will **still** run into nirvana if the fucntion was not found.

Comment (matthiasbeyer) [inline src/objects/object.c:621]: The diff is outdated.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**